Tools/ML/model.h

```diff
@@ -48,8 +48,45 @@ class OnnxModel
 
   // Inferencing
   void initModel(std::string, bool = false, int = 0, uint64_t = 0, uint64_t = 0);
-  float* evalModel(std::vector<Ort::Value>);
-  float* evalModel(std::vector<float>);
+
+  // template methods -- best to define them in header
+  template <typename T>
+  T* evalModel(std::vector<Ort::Value>& input)
+  {
+    LOG(debug) << "Input tensor shape: " << printShape(input[0].GetTensorTypeAndShapeInfo().GetShape());
+    // assert(input[0].GetTensorTypeAndShapeInfo().GetShape() == getNumInputNodes()); --> Fails build in debug mode, TODO: assertion should be checked somehow
+
+    try {
+      auto outputTensors = mSession->Run(mInputNames, input, mOutputNames);
+      LOG(info) << "Number of output tensors: " << outputTensors.size();
```

Comment:
Maybe this could also be changed to LOG(debug)

Reply:
ah, true, I was just printing it for myself locally.